org.eclipse.scanning.api/src/org/eclipse/scanning/api/points/Point.java

```diff
@@ -48,25 +48,35 @@
 	private final String  yName;
 
 	public Point(int xIndex, double xPosition, int yIndex, double yPosition) {
-		this(xIndex, xPosition, yIndex, yPosition, true);
+		this(xIndex, xPosition, yIndex, yPosition, -1, true);
 	}
 
-	public Point(int xIndex, double xPosition, int yIndex, double yPosition, boolean is2D) {
-		this("x", xIndex, xPosition, "y", yIndex, yPosition, is2D);
+	public Point(int xIndex, double xPosition, int yIndex, double yPosition, int stepIndex) {
+		this(xIndex, xPosition, yIndex, yPosition, stepIndex, true);
+	}
+
+	public Point(int xIndex, double xPosition, int yIndex, double yPosition, int stepIndex, boolean is2D) {
+		this("x", xIndex, xPosition, "y", yIndex, yPosition, stepIndex, is2D);
 	}
 
 	public Point(String xName, int xIndex, double xPosition, String yName, int yIndex, double yPosition) {
 		this(xName, xIndex, xPosition, yName, yIndex, yPosition, true);
 	}
 
 	public Point(String xName, int xIndex, double xPosition, String yName, int yIndex, double yPosition, boolean is2D) {
+		this(xName, xIndex, xPosition, yName, yIndex, yPosition, -1, is2D);
+	}
+
+	public Point(String xName, int xIndex, double xPosition, String yName, int yIndex, double yPosition, int stepIndex, boolean is2D) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Constructor has 8 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00107)